src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService.cs

```diff
@@ -118,6 +118,12 @@ protected override bool CanIntroduceVariableFor(ExpressionSyntax expression)
                 return false;
             }
 
+            // (c) For throw expressions.
+            if (expression.IsKind(SyntaxKind.ThrowExpression))
```

Comment:
I think you want to update "CanReplaceWithLValue" instead.  This will also fix up other analyzers/fixers that go through that codepath.

Reply:
Hrmm.. though i suppose if you do that, then it will prevent ExtractMethod from working when you select just the throw-expression (something valuable to test if we don't already have tests).  If that does break ExtractMethod, then it's likely that the approach you're taking here is correct.

Reply:
I didn't understand the suggestion. I don't see `CanReplaceWithLValue` involved in the logic of `IntroduceVariable` and it was not reached when I debugged the tests for this PR.

Reply:
Oh interesting.  it looks like we use that helper for the 'generate local' feature (which is different from the 'introduce local' refactoring).  I was getting them confused.